src/Microsoft.AspNet.Mvc.TagHelpers/ScriptTagHelper.cs

```diff
@@ -22,17 +27,69 @@ namespace Microsoft.AspNet.Mvc.TagHelpers
     /// </remarks>
     public class ScriptTagHelper : TagHelper
     {
+        private const string SrcIncludeAttributeName = "asp-src-include";
+        private const string SrcExcludeAttributeName = "asp-src-exclude";
         private const string FallbackSrcAttributeName = "asp-fallback-src";
+        private const string FallbackSrcIncludeAttributeName = "asp-fallback-src-include";
+        private const string FallbackSrcExcludeAttributeName = "asp-fallback-src-exclude";
         private const string FallbackTestExpressionAttributeName = "asp-fallback-test";
         private const string SrcAttributeName = "src";
 
-        // NOTE: All attributes are required for the ScriptTagHelper to process.
-        private static readonly string[] RequiredAttributes = new[]
-        {
-            FallbackSrcAttributeName,
-            FallbackTestExpressionAttributeName,
+        private static readonly ModeAttributes<Mode>[] ModeDetails = new[] {
+            // Globbed src (include only)
+            ModeAttributes.Create(Mode.GlobbedSrc, new [] { SrcIncludeAttributeName }),
+            // Globbed src (include & exclude)
+            ModeAttributes.Create(Mode.GlobbedSrc, new [] { SrcIncludeAttributeName, SrcExcludeAttributeName }),
+            // Fallback with static src
+            ModeAttributes.Create(
+                Mode.Fallback, new[]
+                {
+                    FallbackSrcAttributeName,
+                    FallbackTestExpressionAttributeName
+                }),
+            // Fallback with globbed src (include only)
+            ModeAttributes.Create(
+                Mode.Fallback, new[] {
+                    FallbackSrcIncludeAttributeName,
+                    FallbackTestExpressionAttributeName
+                }),
+            // Fallback with globbed src (include & exclude)
+            ModeAttributes.Create(
+                Mode.Fallback, new[] {
+                    FallbackSrcIncludeAttributeName,
+                    FallbackSrcExcludeAttributeName,
+                    FallbackTestExpressionAttributeName
+                }),
         };
 
+        private enum Mode
```

Comment:
This is where it is in `LinkTagHelper` and I'm trying to make them as consistent as possible. I can move both, but if at all possible it'd be nice to avoid changing `LinkTagHelper` as part of this work. But, no biggie either way. Just worth comparing stuff in `LinkTagHelper` when reviewing this PR.